src/components/Dialogs/TransactionDialog/TransactionDialog.tsx

```diff
@@ -3,29 +3,85 @@ import ActionDialog, { ActionDialogProps } from '../ActionDialog/ActionDialog'
 import { TextContainer, StyledTransactionIllustration, StyledSpinner } from './TransactionDialog.style'
 import { StyledTitleText, StyledDescriptionText } from '../MessageDialog/MessageDialog.style'
 import { ExtrinsicStatus } from '@/joystream-lib'
+import MessageDialog from '../MessageDialog'
 
 export type TransactionDialogProps = Pick<ActionDialogProps, 'onEnter' | 'onExit' | 'className'> & {
-  onCancel?: () => void
   status: ExtrinsicStatus | null
+  successTitle: string
+  successDescription: string
+  onClose: () => void
```

Comment:
I'm a bit confused. Could we just use `onExit`, here?

Reply:
Probably because `onEnter` and `onExit` may have been confusing. They're not super useful so just removed them. The way they worked, `onExit` would get triggered only when `showDialog` changes from `true` to `false`. But something has to trigger that change, that's why we had `onClose` callback so consumer can react to user closing the dialog